gazebo/rendering/Visual.cc

```diff
@@ -811,9 +811,14 @@ void Visual::SetScale(const ignition::math::Vector3d &_scale)
 
   this->dataPtr->scale = _scale;
 
-  this->dataPtr->sceneNode->setScale(
-      Conversions::Convert(this->dataPtr->scale));
-
+  if (!isnan(this->dataPtr->scale.X()) && !isnan(this->dataPtr->scale.Y())
+      && !isnan(this->dataPtr->scale.Z()))
+  {
+    this->dataPtr->sceneNode->setScale(
+        Conversions::Convert(this->dataPtr->scale));
+  } else {
+    gzerr << Name() << " Scale contains NaNs. Collisions may not visualize properly." << std::endl;
```

Comment:
Is the scale from the `<collision>` in SDF, or is there another source that sets the scale? If it's only from `<collision>`, it would be nice if the error message can say that specifically, so that the user knows what to fix. If I just see scale, I'd go look for a scale tag in the SDF and wouldn't find one.